### DISCUSSION TITLE
Beginner Q/A

### DISCUSSION THREAD

**Discussion by ashwincreates** [xournalpp/xournalpp#3188] [Q&A]
hi everyone,
I am a CS student and I am new to OSS and want to contribute in Xournal. I have used Xournal a lot for my studies and has been very helpful and i want to contribute to it, but since i am new to OSS, working with big projects is kind of tough, do u have any suggestions on where should i start, like i can manage github stuff, but i am having trouble going through the code and understanding the structure, Is there anything i can refer to?

Thanks a lot

**Comment by rolandlo** [comment]
Hi @ashwincreates 

Great to hear that you like Xournal++ and would like to contribute. There is a lot that can be done even if you are new to the code base. First you must find out what you like to do.
- We have some untouched TODO's open in https://github.com/xournalpp/xournalpp/issues/2354 (for instance the one on replacing path string with `fs::path`'s)
- Look at issues list with the [difficulty::easy label](https://github.com/xournalpp/xournalpp/issues?q=is%3Aissue+is%3Aopen+label%3Adifficulty%3A%3Aeasy) or the [Good first issue label](https://github.com/xournalpp/xournalpp/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
- Come up with an own idea how to enhance Xournal++. Let us know beforehand by creating an issue suggesting what you consider to do.

Once you have found out, what you would like to work on, you can just ask us (maintainer team), what files you most likely would have to touch. For a first contribution you will most likely only change few files and don't need to grasp the whole code structure. Feel free to contact us on https://gitter.im/xournalpp/xournalpp

**Reply by ashwincreates** [reply]
Ohk got it, thanks a lot
